scale_build/packages/utils.py

```diff
@@ -0,0 +1,35 @@
+from scale_build.utils.run import run
+from scale_build.utils.paths import DPKG_OVERLAY
+
+
+DEPENDS_SCRIPT_PATH = './scripts/parse_deps.pl'
+
+
+def normalize_bin_packages_depends(depends_str):
+    return list(filter(lambda k: k and '$' not in k, map(str.strip, depends_str.split(','))))
+
+
+def normalize_build_depends(build_depends_str):
+    deps = []
+    for dep in filter(bool, map(str.strip, build_depends_str.split(','))):
+        for subdep in filter(bool, map(str.strip, dep.split('|'))):
+            index = subdep.find('(')
+            if index != -1:
+                subdep = subdep[:index].strip()
+            deps.append(subdep)
+    return deps
+
+
+def get_install_deps(packages, deps, deps_list):
+    for dep in filter(lambda p: p in packages, deps_list):
+        deps.add(packages[dep].source_name)
+        deps.update(
+            get_install_deps(packages, deps, packages[dep].install_dependencies | packages[dep].build_dependencies)
```

Comment:
Why are `build_dependencies` included too?

Reply:
If we want to build package `X` and it has package `C` as a build time dependency and package `C` has package `B` as build time dependency and package `A` as install time dependency, we need `A/B/C` in order to build `X`. Why we need `C`'s build time dependency is that because it's not already built to be available in our mirror and we do need to build it.

Reply:
Then I suppose `get_install_deps` is a misleading name?